Apps/Sandcastle/gallery/development/Custom Primitive.html

```diff
@@ -0,0 +1,327 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta
+      name="viewport"
+      content="width=device-width, initial-scale=1, maximum-scale=1, minimum-scale=1, user-scalable=no"
+    />
+    <meta name="description" content="Custom Primitive example." />
+    <meta name="cesium-sandcastle-labels" content="Development" />
+    <title>Cesium Demo</title>
+    <script type="text/javascript" src="../Sandcastle-header.js"></script>
+    <script
+      type="text/javascript"
+      src="../../../Build/CesiumUnminified/Cesium.js"
+      nomodule
+    ></script>
+    <script type="module" src="../load-cesium-es6.js"></script>
+  </head>
+  <body
+    class="sandcastle-loading"
+    data-sandcastle-bucket="bucket-requirejs.html"
+  >
+    <style>
+      @import url(../templates/bucket.css);
+      #toolbar input[type="range"] {
+        width: 70px;
+      }
+      #toolbar input {
+        vertical-align: middle;
+        padding-top: 2px;
+        padding-bottom: 2px;
+      }
+    </style>
+    <div id="cesiumContainer" class="fullSize"></div>
+    <div id="loadingOverlay"><h1>Loading...</h1></div>
+
+    <script id="cesium_sandcastle_script">
+      function startup(Cesium) {
+        "use strict";
+        //Sandcastle_Begin
+
+        var viewer = new Cesium.Viewer("cesiumContainer");
+
+        /**
+         * Simple example of writing a Primitive from scratch. This
+         * primitive displays a procedurally-generated surface at a given
+         * position on the globe.
+         */
+        function CustomPrimitive(cartographicPosition, context) {
+          this.show = true;
+
+          this.drawCommand = undefined;
+
+          // number of faces wide. There are resolution + 1 vertices.
+          this.faceResolution = 200;
+
+          // Compute a model matrix that puts the surface at a specific point on
+          // the globe.
+          this.cartographicPosition = cartographicPosition;
+          var cartesianPosition = Cesium.Cartographic.toCartesian(
+            cartographicPosition,
+            Cesium.Ellipsoid.WGS84,
+            new Cesium.Cartesian3()
+          );
+          this.modelMatrix = Cesium.Transforms.eastNorthUpToFixedFrame(
+            cartesianPosition,
+            Cesium.Ellipsoid.WGS84,
+            new Cesium.Matrix4()
+          );
+
+          this.halfWidthMeters = 10000;
+          this.boundingSphere = new Cesium.BoundingSphere(
+            cartesianPosition,
+            this.halfWidthMeters * Math.SQRT2
+          );
+
+          this.time = undefined;
+
+          initialize(this, context);
+        }
+
+        /**
+         * generate a quad that's subdivided into faceResolution x faceResolution quads.
+         * The vertices will be adjusted in the vertex shader.
+         */
+        function generateQuad(faceResolution, halfWidthMeters) {
```

Comment:
Minor nitpick - this took me a second because I expected `generateQuad` to generate a single quad. Maybe renaming to `generateVertices` would be clearer?